### PR TITLE
Deprecate and remove all the things for 2.5.0

### PR DESCRIPTION
**Pull request by ktoso**
- udp remoting - now logs warnings if used
- actordsl deprecated
- persistent view removed

I may have missed some migration guides, will fill in on monday.

Refs #21423

### CODE REVIEW

File: akka-docs/rst/java/cluster-usage.rst [hunk 1]
```diff
@@ -610,7 +610,7 @@ Router Example with Pool of Remote Deployed Routees
 
 Let's take a look at how to use a cluster aware router on single master node that creates
 and deploys workers. To keep track of a single master we use the :ref:`cluster-singleton-java`
-in the contrib module. The ``ClusterSingletonManager`` is started on each node.
+in the cluster-tools module. The ``ClusterSingletonManager`` is started on each node.
```
Comment (patriknw) [inline akka-docs/rst/java/cluster-usage.rst]: good catch, same in the scala file

File: akka-contrib/README.md [hunk 2]
```diff
@@ -11,6 +11,16 @@ The conditions under which this transition can occur include:
 If a contributions turns out to not “take off” it may be removed again at a
 later time.
 
+## Deprecation warning
+
+> **Akka-contrib is deprecated** and its contents will be deleted at the next
```
Comment (patriknw) [inline akka-contrib/README.md]: do we need to sprinkle annotations on the classes also?

**Review by patriknw — COMMENTED** [review]
We might have to look closer at what we actually have in contrib. E.g. http://doc.akka.io/docs/akka/2.4/contrib/circuitbreaker.html

File: akka-remote/src/main/resources/reference.conf [hunk 3]
```diff
@@ -572,6 +572,9 @@ akka {
 
     }
 
+    # DEPRECATED
```
Comment (patriknw) [inline akka-remote/src/main/resources/reference.conf]: , since 2.5.0

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst [hunk 4]
```diff
@@ -4,6 +4,31 @@
...
+After being deprecated for a long time, and replaced by :ref:`Persistence Query Java <persistence-query-java>`
+(:ref:`Persistence Query Scala <persistence-query-scala>`) ``PersistentView`` has been removed now removed.
+
+The corresponding query type is ``EventsByPersistenceId``. There are several alternatives for connecting the ``Source``
+to an actor corresponding to a previous ``UntypedPersistentView`` actor:
+
+* `Sink.actorRef`_ is simple, but has the disadvantage that there is no back-pressure signal from the
+  destination actor, i.e. if the actor is not consuming the messages fast enough the mailbox of the actor will grow
+* `mapAsync`_ combined with the ``ask`` pattern is almost as simple with the advantage of back-pressure being propagated all the way
+* `ActorSubscriber`_ in case you need more fine grained control
```
Comment (patriknw) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: don't mention `ActorSubscriber`, perhaps it's better to just link to http://doc.akka.io/docs/akka/2.4/scala/stream/stream-integrations.html#Integrating_with_Actors (and corresponding java docs) than describing it here

**Comment by patriknw** [comment]
Looking good, but I have still not looked at what we have in contrib. Perhaps you should remove that part from this PR to have this one merged quicker.

**Comment by ktoso** [comment]
Ok will do

File: akka-contrib/README.md (hunk 2 above)
Comment (ktoso) [inline akka-contrib/README.md]: yes, but will make contrib a different PR

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst (hunk 4 above)
Comment (ktoso) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: true, changing

**Comment by ktoso** [comment]
Applied all comments, will merge after validation

**Comment by ktoso** [comment]
Contrib will be deprecated in separate PR

**Comment by patriknw** [comment]
Binary compatibility check failed

**Review by patriknw — APPROVED** [review]
LGTM, when it passes validation

**Comment by patriknw** [comment]
There is a section in docs about ActorDSL http://doc.akka.io/docs/akka/2.4.14/scala/actordsl.html#the-actor-dsl
add a deprecation note or just remove the section.

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst [hunk 5]
```diff
@@ -4,6 +4,24 @@
...
+to an actor corresponding to a previous ``PersistentView``. There are several alternatives for connecting the ``Source``
+to an actor corresponding to a previous ``PersistentView`` actor which are documented in :ref:`stream-integrations-scala` 
+for Scala and :ref:`Java <stream-integrations-java>`.
+  
+The consuming actor may be a plain ``Actor`` or an ``PersistentActor`` if it needs to store its own state (e.g. ``fromSequenceNr`` offset).
+
+Please note that Persistence Query is not experimental anymore in Akka ``2.5.0``, so you can safely upgrade to it.
+
 Akka Streams
 ============
```
Comment (patriknw) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: a note about ActorDSL also

**Comment by patriknw** [comment]
LGTM again, after ActorDSL docs
